TreeModelLib/GrowthAndDeathDynamics/SimpleBettina/SimpleBettina.py

```diff
@@ -0,0 +1,148 @@
+        geometry["h_stem"] = self.h_stem
+        print(self.r_crown)
+        print(self.r_stem)
+        print(self.r_root)
+        print(self.h_stem)
+
+        tree.setGeometry(geometry)
+        if (self.die == 0):
+            tree.setSurvival(1)
+        else:
+            tree.setSurvival(0)         
+
+    def treeGrowth(self):
+        inc_rs = (self.weight_girthgrowth * self.grow /
+                (2 * np.pi * self.r_stem * self.flow_length)) 
+        self.r_stem += inc_rs
+        inc_hs = (self.weight_stemgrowth * self.grow /
+                (np.pi * self.r_stem ** 2))
+        inc_rc = (self.weight_crowngrowth * self.grow /
+                (2 * np.pi * (self.h_crown * self.r_crown +
+                self.r_stem ** 2)))
+        inc_rr = (self.weight_rootgrowth * self.grow /
+                (2 * np.pi * self.r_root * self.h_root +
+                 0.5 ** 0.5 * np.pi * self.r_stem ** 2))
+        self.r_root += inc_rr
+        self.h_stem += inc_hs
+        self.r_crown += inc_rc
+ 
+    def treeGrowthWeights(self):
+        self.weight_stemgrowth = (
+                self.parameter["half_max_h_growth_weight"] /
+                (1 + np.exp(- (self.r_crown - self.r_root) /
+                              (self.r_crown + self.r_root) /
+                              self.parameter["h_sigmo_slope"])))
+        self.weight_crowngrowth = (
+                (1 - self.weight_stemgrowth) /
+                (1 + np.exp((self.ag_resources - self.bg_resources) /
+                            (self.ag_resources + self.bg_resources) /
+                            self.parameter["sigmo_slope"])))
+
+        self.weight_girthgrowth = (
+                (1 - self.weight_stemgrowth - self.weight_crowngrowth) /
+                (1 + np.exp((self.root_surface_resistance -
+                             self.xylem_resistance) /
+                 (self.root_surface_resistance + self.xylem_resistance) /
+                 self.parameter["sigmo_slope"])))
+
+        self.weight_rootgrowth = (1 - self.weight_stemgrowth -
+                                  self.weight_crowngrowth -
+                                  self.weight_girthgrowth)
+
+    def treeMaintenance(self):
+        self.maint = self.volume * self.parameter["maint_factor"] * self.time
+
+    def flowLength(self):
+        self.flow_length = (
+                2 * self.r_crown + self.h_stem +
+                0.5 ** 0.5 * self.r_root)
+
+    def treeVolume(self):
+        self.volume = (self.h_root * np.pi * self.r_root**2 +
+                self.flow_length * np.pi * self.r_stem**2 +
+                self.h_crown * np.pi * self.r_crown**2)
+
+    def agResources(self, aboveground_resources):
+        self.ag_resources = aboveground_resources * (
+                np.pi * self.r_crown**2 *
+                self.parameter["sun_c"] * self.time)
+ 
+    def bgResources(self, belowground_resources):
+        self.rootSurfaceResistance()
+        self.xylemResistance()
+        self.deltaPsi()
```

Comment:
you only use delta_psi once. So it does not need to be stored to long
```suggestion
        delta_psi = self.deltaPsi()
```
and see in function deltaPsi

Reply:
jaja!